lib/IRGen/GenStruct.cpp

```diff
@@ -680,6 +720,12 @@ class ClangRecordLowering {
 
   const TypeInfo *createTypeInfo(llvm::StructType *llvmType) {
     llvmType->setBody(LLVMFields, /*packed*/ true);
+    if (SwiftType.getStructOrBoundGenericStruct()
+            ->isCxxNotTriviallyCopyable()) {
+      return ClangRecordAddressOnlyTypeInfo::create(
+          FieldInfos, llvmType, TotalStride, std::move(SpareBits),
```

Comment:
`ClangRecordLowering::addStructField` assumes that all fields will have `LoadableTypeInfo` (by invoking `cast<LoadableTypeInfo>()`) -- but that's no longer true starting with this PR. Are you getting any assertions in the testsuite when processing types that aggregate other types which we consider non-trivial?

Reply:
Interesting... you're right. No tests caught this so, I need to add a few. I think we would have hit an assertion if we had the tests from #32291. I changed `LoadableTypeInfo` to `FixedTypeInfo` which seemed to work for everything except `getExplosionSize` which I was able to replace with `SwiftDecl->getStoredProperties().size()` for non-loadable types.

Reply:
I verified the tests I added triggered an error with the old code (`cast<LoadableTypeInfo>()`).